lib/ansible/plugins/filter/ipaddr.py

```diff
@@ -767,6 +770,23 @@ def ipsubnet(value, query='', index='x'):
                     return str(value.supernet(query)[0])
                 except:
                     return False
+    elif str(query):
```

Comment:
Please move this to a var before if/elif/else branching to prevent stringify operation on each and every check.